src/gluonts/model/seq2seq/_forking_estimator.py

```diff
@@ -369,51 +378,90 @@ def create_transformation(self) -> Transformation:
-            quantile_output=trained_network.quantile_output,
-            context_length=self.context_length,
-            cardinality=self.cardinality,
-            embedding_dimension=self.embedding_dimension,
-            scaling=self.scaling,
-            scaling_decoder_dynamic_feature=self.scaling_decoder_dynamic_feature,
-            dtype=self.dtype,
-        )
+            prediction_network = ForkingSeq2SeqPredictionNetwork(
+                encoder=trained_network.encoder,
+                enc2dec=trained_network.enc2dec,
+                decoder=trained_network.decoder,
+                quantile_output=trained_network.quantile_output,
+                context_length=self.context_length,
+                cardinality=self.cardinality,
+                embedding_dimension=self.embedding_dimension,
+                scaling=self.scaling,
+                dtype=self.dtype,
+            )
 
-        copy_parameters(trained_network, prediction_network)
+            copy_parameters(trained_network, prediction_network)
 
-        return RepresentableBlockPredictor(
-            input_transform=transformation,
-            prediction_net=prediction_network,
-            batch_size=self.trainer.batch_size,
-            freq=self.freq,
-            prediction_length=self.prediction_length,
-            ctx=self.trainer.ctx,
-            forecast_generator=QuantileForecastGenerator(quantile_strs),
-        )
+            return RepresentableBlockPredictor(
+                input_transform=transformation,
+                prediction_net=prediction_network,
+                batch_size=self.trainer.batch_size,
+                freq=self.freq,
+                prediction_length=self.prediction_length,
+                ctx=self.trainer.ctx,
+                forecast_generator=QuantileForecastGenerator(quantile_strs),
+            )
+        else:
+            assert self.distr_output is not None
+            prediction_network = ForkingSeq2SeqDistributionPredictionNetwork(
+                encoder=trained_network.encoder,
+                enc2dec=trained_network.enc2dec,
+                decoder=trained_network.decoder,
+                distr_output=trained_network.distr_output,
+                context_length=self.context_length,
+                cardinality=self.cardinality,
+                embedding_dimension=self.embedding_dimension,
+                scaling=self.scaling,
+                dtype=self.dtype,
+            )
+
+            copy_parameters(trained_network, prediction_network)
+
+            return RepresentableBlockPredictor(
+                input_transform=transformation,
+                prediction_net=prediction_network,
+                batch_size=self.trainer.batch_size,
+                freq=self.freq,
+                prediction_length=self.prediction_length,
+                ctx=self.trainer.ctx,
+                forecast_generator=DistributionForecastGenerator(
+                    self.distr_output
+                ),
```

Comment:
We could factor generator out as a variable, and set it in the appropriate if else cast, and then just pass the variable to the forecast_generator. This way we have less code duplication, since both predictors are wrapped in a RepresentableBlockPredictor.

Reply:
addressed